core/src/main/java/org/infinispan/protostream/impl/JsonUtils.java

```diff
@@ -280,6 +285,76 @@ private static void processObject(ImmutableSerializationContext ctx, JsonParser
       writer.flush();
    }
 
+   private static void processMap(ImmutableSerializationContext ctx, MapDescriptor md, JsonParser parser, TagWriter writer) throws IOException {
+      while (true) {
+         JsonToken token = parser.nextToken();
+         if (token == JsonToken.END_OBJECT) {
+            break;
+         }
+         if (token != JsonToken.FIELD_NAME) {
+            throw new IllegalStateException("Unexpected token");
+         }
+         ByteArrayOutputStream baos = new ByteArrayOutputStream(ProtobufUtil.DEFAULT_ARRAY_BUFFER_SIZE);
+         TagWriter nestedWriter = TagWriterImpl.newInstanceNoBuffer(ctx, baos);
+         String key = parser.getCurrentName();
+         switch (md.getKeyType()) {
+            case STRING -> nestedWriter.writeString(1, key);
+            case INT32 -> nestedWriter.writeInt32(1, Integer.parseInt(key));
+            case INT64 -> nestedWriter.writeInt64(1, Long.parseLong(key));
+            case FIXED32 -> nestedWriter.writeFixed32(1, Integer.parseInt(key));
+            case FIXED64 -> nestedWriter.writeFixed64(1, Long.parseLong(key));
+            case SINT32 -> nestedWriter.writeSInt32(1, Integer.parseInt(key));
+            case SINT64 -> nestedWriter.writeSInt64(1, Long.parseLong(key));
+            case SFIXED32 -> nestedWriter.writeSFixed32(1, Integer.parseInt(key));
+            case SFIXED64 -> nestedWriter.writeSFixed64(1, Long.parseLong(key));
+            case UINT32 -> nestedWriter.writeUInt32(1, Integer.parseInt(key));
+            case UINT64 -> nestedWriter.writeUInt64(1, Long.parseLong(key));
```

Comment:
isn't double/float missing?

Reply:
Those are not valid protobuf map keys. Only scalars are supported